tests/components/recorder/test_purge.py

```diff
@@ -55,6 +55,23 @@ def _add_test_states(self):
                     event_id=event_id + 1000
                 ))
 
+            # add a state that is old but the only state of its entity and
+            # should be protected
+            if hasattr(self, "_protected_event_id"):
```

Comment:
I don't understand this if-check. What code inside this for loop can change the outcome of this if/else statement?

Also, you can use `getattr(self, "_protected_event_id", 2000)` to supply a default value